cairo_jpg.c

```diff
@@ -91,18 +93,17 @@ cairo_status_t cairo_surface_write_to_jpeg_stream(cairo_surface_t *sfc,
 		cairo_write_func_t write_func, void *closure, int quality) {
 	cairo_status_t e;
 	unsigned char *data = NULL;
-	size_t len = 0;
+	unsigned long len = 0;
 
 	e = cairo_surface_write_to_jpeg_mem(sfc, &data, &len, quality);
-	if (e != CAIRO_STATUS_SUCCESS) {
-		return e;
+	if (e == CAIRO_STATUS_SUCCESS) {
+		assert(sizeof(unsigned long) <= sizeof(size_t)
```

Comment:
Hmm, I don't get why this assertion is still required. `len` is `unsigned long` and `cairo_write_func_t` takes an  `unsigned long`… Am I missing something?